src/engine/guidance/post_processing.cpp

```diff
@@ -812,6 +816,11 @@ void collapseTurnAt(std::vector<RouteStep> &steps,
 //
 bool isStaggeredIntersection(const RouteStep &previous, const RouteStep &current)
 {
+    //don't touch roundabouts
+    if (entersRoundabout(previous.maneuver.instruction) ||
+        entersRoundabout(current.maneuver.instruction))
+        return false;
+
```

Comment:
The check is doing exactly that, it disables the staggered intersections for roundabouts? It makes sure that wherever it is checked, staggered intersections don't interact with roundabouts. So what's your question here?